### PR TITLE
Add support for a file listing expected package to objc prefixes for validation.

### PR DESCRIPTION
**Pull request by thomasvl**
- Add a env var to pass a set of expected prefixes for validation.
- Report warnings/errors based on the expected prefixes vs. the data in the files compiled.
- Use some helpers from common directory.

### CODE REVIEW

File: objectivec/README.md [hunk 1]
```diff
@@ -133,6 +133,21 @@ be collisions. This option provides a prefix that will be added to the Enums
...
 the prefix on the package the proto is in.
 
+**GPB_OBJC_EXPECTED_PACKAGE_PREFIXES** (no default)
+
+This environment variable can be set when invoking protoc to point to a file
+that lists expected `package` to `objc_class_prefix` values.  The compiler will
+then issue warnings and/or errors if the data in the proto files being generated
+is not listed in the expectations file. The format for the file is:
+
+- An entry is done via `package=prefix`.
```
Comment (pherl) [inline objectivec/README.md]: Can we use something like protobuf text format instead? This avoid creating a parser for the configurations.

Also before this change, generators are deterministic solely on the parsed descriptor protos. This change will add additional input to the protoc. I'd rather explore other options before committing to this approach.

**Comment by TeBoring** [comment]
For this PR, there is still one small concern: is it to possible to make the introduced macro not visible to normal developers? Or at least change the name to infer that should not be used by normal developers?

**Comment by thomasvl** [comment]
Which macro?  I didn't think I introduced a macro.  If you mean the env variable, devs should be able to find/use it since it helps ensure their proto files are consistent with respect to package -- objc prefix usage.

**Comment by TeBoring** [comment]
For now, can we not mention this in README? Because the config file format and the way to trigger this feature (env, protoc argument, plugin argument) is subject to change in future.

**Comment by thomasvl** [comment]
Removed from the README

**Comment by TeBoring** [comment]
LGTM

**Comment by thomasvl** [comment]
I guess the jruby tests are still flaky but not marked as such?  I can't say I see how my cl failed them.

**Comment by thomasvl** [comment]
fyi - the change passed when i pushed it to my repo - https://travis-ci.org/thomasvl/protobuf/builds/75640395

**Pull request merged**